exporters/otlp/otlpmetric/internal/otlpconfig/options.go

```diff
@@ -72,24 +74,48 @@ type (
 	}
 )
 
-func NewDefaultConfig() Config {
-	c := Config{
+// NewHTTPConfig returns a new Config with all settings applied from opts and
+// any unset setting using the default HTTP config values.
+func NewHTTPConfig(opts ...HTTPOption) Config {
+	cfg := Config{
 		Metrics: SignalConfig{
-			Endpoint:    fmt.Sprintf("%s:%d", DefaultCollectorHost, DefaultCollectorPort),
+			Endpoint:    fmt.Sprintf("%s:%d", DefaultCollectorHost, DefaultCollectorHTTPPort),
 			URLPath:     DefaultMetricsPath,
 			Compression: NoCompression,
 			Timeout:     DefaultTimeout,
 		},
 		RetryConfig: retry.DefaultConfig,
 	}
+	cfg = ApplyHTTPEnvConfigs(cfg)
+	for _, opt := range opts {
+		cfg = opt.ApplyHTTPOption(cfg)
+	}
 
-	return c
+	tmp := strings.TrimSpace(cfg.Metrics.URLPath)
```

Comment:
(probably for another PR): this is repeated in the trace exporter as well. How about a shared `CleanPathOr(path, default)` method?

Reply:
I'm in favor of the unification of this shared functionality, but I do not want to include it in this PR. This PR is intended to primarily resolve #2614, adding such unification would overload the PR.